bennu-toolkit/src/main/java/org/fenixedu/bennu/toolkit/components/Component.java

```diff
@@ -0,0 +1,66 @@
+package org.fenixedu.bennu.toolkit.components;
+
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+
+import org.fenixedu.commons.i18n.LocalizedString;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+
+public abstract class Component {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(Component.class);
+
+    private static Map<String, Component> COMPONENTS = Maps.newHashMap();
+
+    public abstract Element process(Element element);
+
+    public static Component getComponent(String key) {
+        return Component.COMPONENTS.get(key);
+    }
+
+    public static Collection<Component> getComponents() {
+        return COMPONENTS.values();
+    }
+
+    public static String process(String origin) {
+        Document doc = Jsoup.parse(origin);
+        Elements components = doc.select("[bennu-component]");
+
+        for (Element component : components) {
+            String key = component.attr("bennu-component");
+            Element result = COMPONENTS.get(key).process(component);
```

Comment:
You should gracefully handle a definition of a missing component. This way it will just NPE.